### DISCUSSION TITLE
Bind PocketBase to localhost only and serve domain at the same time

### DISCUSSION THREAD

**Discussion by macnux** [pocketbase/pocketbase#6211] [Q&A] [answered]
Hi,

Can I run pocketbase in production `pocketbase serve yourdomain.com` and at the same time bind PocketBase to localhost only using `--http="127.0.0.1:8090"`?
So the command will be:
`pocketbase serve mydomain.com --http="127.0.0.1:8090`
So no one can access it from outside.

Regards,

**Comment by ganigeorgiev** [comment]
No, unless I'm misunderstanding what you are asking.

When you specify a domain to the `serve` command, PocketBase issues a Let's Encrypt certificate and a public access is required at least for the http01/dns01 challenge validations.

You've written _"no one can access it from outside"_ but why do you need a domain in the first place? What is actually you are trying to do? If you want a self-signed certificate PocketBase cannot generate it for your out of the box.

If you just want a host address for you service then you can simply add one in your `/etc/hosts` file and map it to localhost. You can either bind PocketBase to `127.0.0.1:port` OR if you want to bind to all IPv4 interfaces you can use `0.0.0.0:port`.

For any other more advanced network configuration you'll have to use a reverse-proxy like nginx, apache, caddy, etc. (_there is also option to modify the default Go Server handler using the `OnServe` hook but without any details or context about your project I can't recommend it_).

**Reply by macnux** [reply]
Okay, if I bind it locally, will this affect any functionality?
Since the frontend will be ssl domain and will call this api auth and other collections from a non ssl endpoint.

**Reply by thangngoc89** [reply, marked as answer]
@macnux it wouldn’t work. Binding to localhost means you could only access it on the same computer unless you have a proxy in front.

**Reply by thangngoc89** [reply]
@macnux just a side note

> Since the frontend will be ssl domain and will call this api auth and other collections from a non ssl endpoint.

This wouldn't work either. It would cause "mixed content error", meaning you can't call non-SSL endpoints from a SSL site.

**Reply by macnux** [reply]
So it's a must to make api endpoint public to be able to work with it from ssl frontend?
I have to use a firewall to block access for its port so no one can access from outside

**Reply by thangngoc89** [reply]
@macnux I think you’re confusing about lots of terms here. 
Let’s not talk about SSL because it only adds confusion to the matter. 

If you want the frontend (browsers) to access pocketbase, you need to expose the port, either binding to 0.0.0.0:port or you use a reverse proxy like nginx to handle the exposure. Either way, there must be an open port in your server for browser to communicate with pocketbase’s API. 

Now let’s add SSL to the matter, the port for http (without SSL) is 80, the port for https (with SSL) is 443.

**Reply by macnux** [reply]
I'd like to explain my idea for further understanding.
My frontend is next.js app and I use nginx on the server (I'm using centos webpanel).
The backend is pocketbase and both are hosted on the same server.
I want only the next.js app to access pocketbase and no one can access the API endpoints itself. Just the nextjs only.
That's why I ask about if I bind it locally, does this affect using pocketbase auth and other stuff?

**Reply by thangngoc89** [reply]
@macnux There you go. You should have provided all the context in the beginning (the purpose instead of the method). I personally don't touch next.js, I only develop SPA using React so I don't know for sure. But there could be 2 situations:

- next.js server <-> pocketbase server communication: for sure you could just bind to localhost and let's the next.js handle the API
- next.js client <-> pocketbase server, if this happens then you need to expose it. 

Another way is you could use nginx and proxy these routes: `/api` and `/_/` requests to the pocketbase server so you don't have to expose the port.

Finally, please read [the SSR intergration guide and gotchas](https://github.com/pocketbase/pocketbase/discussions/5313) from @ganigeorgiev in case you didn't read that
